chia/full_node/fee_tracker.py

```diff
@@ -61,6 +61,16 @@ def get_estimate_time_intervals() -> List[uint64]:
     return [uint64(blocks * SECONDS_PER_BLOCK) for blocks in get_estimate_block_intervals()]
 
 
+def get_bucket_index(sorted_buckets: SortedDict, fee_rate: float) -> int:
+    if fee_rate in sorted_buckets:
+        bucket_index = sorted_buckets[fee_rate]
+    else:
+        # Choose the bucket to the left if we do not have exactly this fee rate
+        bucket_index = sorted_buckets.bisect_left(fee_rate) - 1
```

Comment:
it seems like it would be simpler and faster to *just* use `bisect_left()`. your keys are floats after all, an exact hit is presumably unlikely, no?

Reply:
why do you need to do `- 1` btw? presumably `bisect_left()` will give you the first element.
It seems there would be a risk you end up with index `-1` here

Reply:
It's funny you mention it. The NEXT PR after this one fixes that and adds tests.

I just want to remove this refactor from the actual bugfix commit.

I would prefer to put this refactor in first, and then the bugfix - let me know which way you want me to go.

Reply:
Yes. This patch is a prelude to a fix for that. Take a look at https://github.com/Chia-Network/chia-blockchain/pull/14270#discussion_r1061940380